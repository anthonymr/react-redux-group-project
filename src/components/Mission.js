import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { reserveMission, leaveMission } from '../redux/missions/missionSlice';
import style from '../styles/Mission.module.css';

function Mission({
  name, description, id, reserved,
}) {
  const dispatch = useDispatch();

  function reserveClickHandler() {
    dispatch(reserveMission(id));
  }

  function leaveClickHandler() {
    dispatch(leaveMission(id));
  }

  return (
    <tr className={style.missionstablerow}>
      <td className={style.missionstablename}>{name}</td>
      <td>{description}</td>
      <td>
        {
          !reserved
          && <span className={style.labelNotMember}>NOT A MEMBER</span>
        }
        {
          reserved
          && <span>Active Member</span>
        }
      </td>
      <td>
        {
          !reserved
          && <button type="button" onClick={reserveClickHandler} className={style.joinbutton}>Join Mission</button>
        }
        {
          reserved
          && <button type="button" onClick={leaveClickHandler} className={style.leavebutton}>Leave Mission</button>
        }
      </td>
    </tr>
  );
}

Mission.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  reserved: PropTypes.bool,
};

Mission.defaultProps = {
  reserved: false,
};

export default Mission;
