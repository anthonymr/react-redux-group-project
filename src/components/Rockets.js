import { useSelector } from 'react-redux';
import Rocket from './Rocket';

function Rockets() {
  const { rockets } = useSelector((state) => state.rockets);

  return (
    <section>
      <ul>
        {
          rockets.map((rocket) => (
            <li key={rocket.id}>
              <Rocket flickrImages={rocket.flickr_images} name={rocket.name} description={rocket.description} />
            </li>
          ))
        }
      </ul>
    </section>
  );
}

export default Rockets;
