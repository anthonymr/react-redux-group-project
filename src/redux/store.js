import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketsReducer,
  },
});

export default store;
