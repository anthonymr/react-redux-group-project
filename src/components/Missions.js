import { useSelector } from 'react-redux';
import Mission from './Mission';
import style from '../styles/Missions.module.css';

function Missions() {
  const { missions } = useSelector((state) => state.missions);

  return (
    <section>
      <table className={style.missionstable}>
        <thead className={style.missionstablehead}>
          <tr className={style.missionstablerow}>
            <th scope="col" style={{ width: '10%' }}>Mission</th>
            <th scope="col" style={{ width: '60%' }}>Description </th>
            <th scope="col" style={{ width: '15%' }}>Status</th>
            <th scope="col" style={{ width: '15%' }}>Join</th>
          </tr>
        </thead>
        { missions.map((mission) => (
          <tbody className={style.tablebody} key={mission.id}>
            <Mission
              name={mission.name}
              description={mission.description}
              id={mission.id}
              reserved={mission?.reserved}
            />
          </tbody>
        ))}
      </table>
    </section>
  );
}

export default Missions;
