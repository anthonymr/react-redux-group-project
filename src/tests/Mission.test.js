import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Missions from '../components/Missions';
import Rockets from '../components/Rockets';
import Mission from '../components/Mission';

it('renders correctly', () => {
  const mockedMission = {
    id: 'testingMission',
    name: 'testingMission',
    description: 'testingDescription',
    reserved: false,
  };

  const tree = renderer
    .create(
      <Provider store={store}>
        <Missions />
        <Rockets />
        <Mission
          name={mockedMission.name}
          description={mockedMission.description}
          id={mockedMission.id}
          reserved={mockedMission?.reserved}
        />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
