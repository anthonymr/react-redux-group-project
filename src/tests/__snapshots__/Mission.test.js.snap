// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <section>
    <table
      className="missionstable"
    >
      <thead
        className="missionstablehead"
      >
        <tr
          className="missionstablerow"
        >
          <th
            scope="col"
            style={
              Object {
                "width": "10%",
              }
            }
          >
            Mission
          </th>
          <th
            scope="col"
            style={
              Object {
                "width": "60%",
              }
            }
          >
            Description 
          </th>
          <th
            scope="col"
            style={
              Object {
                "width": "15%",
              }
            }
          >
            Status
          </th>
          <th
            scope="col"
            style={
              Object {
                "width": "15%",
              }
            }
          >
            Join
          </th>
        </tr>
      </thead>
    </table>
  </section>,
  <section
    className="rocketsContainer"
  >
    <ul
      className="rocketList"
    />
  </section>,
  <tr
    className="missionstablerow"
  >
    <td
      className="missionstablename"
    >
      testingMission
    </td>
    <td>
      testingDescription
    </td>
    <td>
      <span
        className="labelNotMember"
      >
        NOT A MEMBER
      </span>
    </td>
    <td>
      <button
        className="joinbutton"
        onClick={[Function]}
        type="button"
      >
        Join Mission
      </button>
    </td>
  </tr>,
]
`;
